backend/server/controllers/courseinstances.js

```diff
@@ -69,6 +23,72 @@ module.exports = {
       .catch(error => res.status(400).send(error))
   },
 
+  testi(req, res) {
+    const errors = []
```

Comment:
even if this shit is full of ifs now... it can be split into smaller pieces and into controller helpers.